Sources/ScipioKit/Producer/FrameworkProducer.swift

```diff
@@ -141,22 +120,93 @@ struct FrameworkProducer {
         }
     }
 
-    private func restoreAllAvailableCaches(
+    private func restoreAllAvailableCachesIfNeeded(
         availableTargets: Set<CacheSystem.CacheTarget>,
         cacheSystem: CacheSystem
     ) async -> Set<CacheSystem.CacheTarget> {
-        let chunked = availableTargets.chunks(ofCount: cacheStorage?.parallelNumber ?? CacheSystem.defaultParalellNumber)
+        let cacheStorages: [any CacheStorage]
+
+        switch cacheMode {
+        case .disabled:
+            return []
+        case .project:
```

Comment:
It's very confusing `nil` means project cache.

How about adding `ProjectCacheStorage` to represent project cache mode?

By adding this, we can represent `.project` as `.storages([projectCacheStorage, [consumer, producer]])`